operator/v1/types_ingress.go

```diff
@@ -225,6 +232,97 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// SecuritySpec defines the settings for securing IngressController connections.
+type SecuritySpec struct {
+	// Profile is one of three possible security configurations from [1]:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// If unset, the default is "Intermediate".
+	//
+	// +optional
+	Profile *SecurityProfileType `json:"profile,omitempty"`
+	// CustomSettings defines the schema for customizing the security settings of
+	// IngressController connections.
+	//
+	// +optional
+	CustomSettings *CustomProfileSettings `json:"customSettings,omitempty"`
```

Comment:
`SecuritySpec` seems a lot like a union type. If so, what about a new `SecurityProfileType` named "Custom"? Then:

```go
type SecuritySpec struct {
  Profile *SecurityProfileType
  // Custom is a user defined profile with defaults based on the Intermediate profile.
  Custom *CustomSecurityProfile
}
```

Reply:
The intent is to allow only a `Profile` of type "Custom" to specify `CustomSettings` fields. All other `Profile` types will not configurable or have individual settings overridden.

Reply:
Custom's relationship with Intermediate still makes the word "custom" somewhat weird to me. Custom seems intrinsically bound to Intermediate, which says to me the thing you're really configuring is the Intermediate profile. Like:

```go
type SecuritySpec struct {
  Profile *SecurityProfileType

  Intermediate *SecurityProfile
}

type SecurityProfile struct {
  Ciphers
  SecurityProtocol
}
```

This also relates to another thing I wanted to mention, which is that I expected a new field on `IngressControllerStatus` showing the effective profile. The `SecurityProfile` struct above could be embedded in status.

Reply:
> This also relates to another thing I wanted to mention, which is that I expected a new field on IngressControllerStatus showing the effective profile. The SecurityProfile struct above could be embedded in status.

Good point. I will add `Profile` to status.